global-script.lua

```diff
@@ -3245,4 +3410,282 @@ function tCompare(t1,t2)
         return table.concat(newTab,"|")
     end
     return cc2(t1) == cc2(t2)
+end
+
+function swapPlayerAreaColors(a, b)
+    if a == b then return end
+    local function tableSwap(table)
+        table[a], table[b] = table[b], table[a]
+    end
+    local function tintSwap(table)
+        local oa = table[a]
+        local ob = table[b]
+        local ta = oa.getColorTint()
+        local tb = ob.getColorTint()
+        oa.setColorTint(tb)
+        ob.setColorTint(ta)
+        tableSwap(table)
+    end
+    local function positionSwap(table)
+        local oa = table[a]
+        local ob = table[b]
+        if type(oa) == "string" then
+            oa = getObjectFromGUID(oa)
+            ob = getObjectFromGUID(ob)
+        end
+        local ta = oa.getPosition()
+        local tb = ob.getPosition()
+        oa.setPosition(tb)
+        ob.setPosition(ta)
+    end
+    local function handSwap(i)
+        local ta = Player[a].getHandTransform(i)
+        local tb = Player[b].getHandTransform(i)
+        Player[a].setHandTransform(tb, i)
+        Player[b].setHandTransform(ta, i)
+    end
+    local function updateBlock(color)
+        local o = playerBlocks[color]
+        if not o then
```

Comment:
is this more a sanity check? because playerBlocks should always have a value right?

Reply:
`playerBlocks["White"]` does not exist, for instance, but you can change to be the white player.  But there's actually a bug right below that (I use both Pink and Teal as temporary colors, and it's supposed to be the same color in both places), which is now fixed.

Reply:
Said Pink/Teal bug is now fixed.